### DISCUSSION TITLE
How to package a Windows app from the command line

### DISCUSSION THREAD

**Discussion by hamada-applicita** [unoplatform/uno#13471] [Q&A] [answered]
I am starting with a simple Uno Windows desktop application.

I can successfully sign, version and package my application using the Visual Studio 2022 UI. (Right click the windows project and select 'Package and Publish')

All my attempts at doing this from the command line (so I can auotmate the process) have failed. I am running

`msbuild "UnoApp1.sln" /p:AppxBundlePlatforms="x86|x64" /p:AppxBundle=Always /p:UapAppxPackageBuildMode=StoreUpload /p:AppxPackageDir="C:/Code/AppxPackages" /p:AppxPackageSigningEnabled=false`

I dont get any errors but no msix file is being created. What I am missing?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. The official documentation related to WinAppSDK publishing can be found [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-packaged-apps) and [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp), including a github action sample.

**Reply by hamada-applicita** [reply]
Thanks, I found the missing command line parameter from here - https://github.com/andrewleader/WindowsAppSDKGallery/blob/main/.github/workflows/dotnet-desktop.yml#L102

`msbuild "UnoApp1.sln" /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64" /p:AppxBundle=Always /p:UapAppxPackageBuildMode=StoreUpload /p:AppxPackageDir="C:/Code/AppxPackages/" /p:AppxPackageSigningEnabled=false`
